tools/c7n_mailer/c7n_mailer/gcp/pubsub_subscriptions.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from googleapiclient.errors import HttpError
+import base64
+import zlib
+from datetime import tzinfo, timedelta
+
+MAX_MESSAGES = 200
+
+
+class PubSubSubscriptions(object):
+    """
+    Receive messages from GCP Pub/Sub Service:
+    C7N_GCP-Mailer Processor class uses this utility class to connect to subscriptions
+    to receive messages.
+    """
+
+    @staticmethod
+    def receive_messages(session, subscription, max_messages=None):
+        """Receive messsage(s) from subscribed topic
+        """
+        subscriber_param = PubSubSubscriptions.get_subscription_param(session, subscription)
+
+        client = session.client('pubsub', 'v1', 'projects.subscriptions')
```

Comment:
so another way to do this would be to make an `__init__()` function for this class and instantiate the client as a class member. That would save a new client having to be opened for every method call (I'm not for sure if it's actually connecting to a server or not on instantiation. If it wasn't, then it might not be a big deal).

However at a minimum if you kept it just the way it was in each method, I would wrap it in a protected method (e.g. `_create_client()`, or make the parameters class variables that are passed in:
```
self.a = pubsub
self.b = v1
self.c = proj.subscriptions

session.client(a,b,c)
```

That way when one of those needs to change like the version, you just have to update in one place in either option listed above.